text/0000-format-args-implicit-identifiers.md

```diff
@@ -0,0 +1,340 @@
+
+    println!("hello {get_person()}");  // function call
+    println!("hello {self.person}");   // field access
+
+The RFC author anticipates in particular that field access may be requested by many as part of this RFC. After careful consideration this RFC does not propose to go further than the single identifier special case, proposed above as implicit named arguments.
+
+If any expressions beyond identifiers become accepted in format strings, then the RFC author expects that users will inevitably ask "why is *my* particular expression not accepted?". This could lead to feature creep, and before long perhaps the following might become valid Rust:
+
+    println!("hello { if self.foo { &self.person } else { &self.other_person } }");
+
+This no longer seems easily readable to the RFC author.
+
+### Proposed Interpolation Syntax
+
+Early review of this RFC raised an observation that the endpoint of such feature creep would be that eventually Rust would embrace interpolation of any expressions inside these macros.
+
+To keep interpolation isolated from named and positional arguments, as well as for readability and (possibly) to reduce parsing complexity, curly-plus-bracket syntax was proposed for interpolation:
+
+    println!("hello {(get_person())}");
+    println!("hello {(self.person)}");
+
+Indeed the RFC's perverse example reads slightly easier with this syntax:
+
+    println!("hello {( if self.foo { &self.person } else { &self.other_person } )}");
+
+Because the interpolation syntax `{(expr)}` is orthogonal to positional `{}` and named `{ident}` argument syntax, and is a superset of the functionality which would be offered by implict named arguments, the argument was made that we should make the leap directly to interpolation without introducing implicit named arguments so as to avoid complicating the existing cases.
+
+### Argument Against Interpolation
+
+It should first be noted that the interpolation in other languages is often a language feature; if they have string formatting functions they typically do not enjoy syntax-level support. Instead other language formatting functions often behave similarly to Rust's positional and/or named arguments to formatting macros.
+
+For example, Python 3's `.format()` method is on the surface extremely similar to Rust's formatting macros:
+
+    "hello {}".format(person)
+    "hello {person}".format(person=person)
+
+However, Python 3 cannot improve the ergonomics of these functions in the same way that this RFC proposes to use implicit named arguments. This is for technical reasons: Python simply does not have a language mechanism which could be used to add implicit named arguments to the `.format()` method. As a result, offering improved ergonomics in Python would neccesitate the introduction of a language-level interpolation syntax.
+
+(Note, the closest Python 3's `.format()` can get to implicit named arguments is this:
```

Comment:
As of python 3.6, python supports literal interpolation strings suing the `f` prefix. So you can do something like `f"{person}"`. Of course that is a language-level interpolation syntax, but saying this use for format is the closest python gets isn't quite accurate.

Reply:
Thanks, I have adjusted the previous paragraph to make a clear reference to the prior art section where `f"{person}"` in Python 3 is discussed!